javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/RemappingUrlStreamHandler.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+class RemappingUrlStreamHandler extends URLStreamHandler {
+  private final JarFile delegateJarFile;
+
+  public RemappingUrlStreamHandler(URL delegateJarFileLocation) {
+    try {
+      delegateJarFile = new JarFile(new File(delegateJarFileLocation.toURI()), false);
+    } catch (URISyntaxException | IOException e) {
+      throw new IllegalStateException("Unable to read internal jar", e);
+    }
+  }
+
+  /** {@inheritDoc} */
+  protected URLConnection openConnection(URL url) {
+    try {
+      String file = url.getFile();
+      if ("/".equals(file)) {
+        // "/" is used as the default url of the jar
+        // This is called by the SecureClassLoader trying to obtain permissions
+        // nullInputStream() is not available until Java 11
+        return new InputStreamUrlConnection(url, new ByteArrayInputStream(new byte[0]), 0);
+      }
+
+      if (file.startsWith("/")) {
+        file = file.substring(1);
+      }
+      JarEntry entry = delegateJarFile.getJarEntry(file);
+      if (entry == null) {
+        return null;
```

Comment:
I suspect that this should throw an exception instead of returning null to be more in line with url stream handlers from jdk.

Reply:
Throwing `IOException`